store/tikv/txn.go

```diff
@@ -431,19 +451,19 @@ func (txn *tikvTxn) LockKeys(ctx context.Context, lockCtx *kv.LockCtx, keysInput
 			txn.committer.ttlManager.run(txn.committer, lockCtx)
 		}
 	}
-	txn.lockKeys = append(txn.lockKeys, keys...)
 	for _, key := range keys {
+		valExists := kv.SetKeyLockedValueExists
 		// PointGet and BatchPointGet will return value in pessimistic lock response, the value may not exist.
 		// For other lock modes, the locked key values always exist.
 		if lockCtx.ReturnValues {
 			val, _ := lockCtx.Values[string(key)]
-			valExists := len(val.Value) > 0
-			txn.lockedMap[string(key)] = valExists
-		} else {
-			txn.lockedMap[string(key)] = true
+			if len(val.Value) == 0 {
+				valExists = kv.SetKeyLockedValueNotExists
+			}
 		}
+		memBuf.UpdateFlags(key, kv.SetKeyLocked, kv.DelNeedCheckExists, valExists)
```

Comment:
Maybe we need to reset some already locked keys which have check exist needs. It's statement level and maybe there will be some flags left?

Reply:
I have done three flag changes in this `UpdateFlags`:
1. Mark key is locked
2. Delete need check exists flag
3. Record the existence of value